### PR TITLE
fix(weave): Elide display names that exceed the char limit

### PR DESCRIPTION
**Pull request by andrewtruong**
This PR elides all call `display_name` to be at most the max char limit.

Internal thread:
https://weightsandbiases.slack.com/archives/C01T8BLDHKP/p1732213159962419?thread_ts=1732208270.722299&cid=C01T8BLDHKP

### CODE REVIEW

File: weave/trace/util.py [hunk 1]
```diff
@@ -8,6 +8,33 @@
 from threading import Thread as _Thread
 from typing import Any, Callable
 
+LOG_ONCE_MESSAGE_SUFFIX = " (subsequent messages of this type will be suppressed)"
```
Comment (andrewtruong) [inline weave/trace/util.py:11]: This code is moved from `op_extensions/log_once.py` so it's more accessible

File: weave/trace/op.py [hunk 2]
```diff
@@ -29,8 +29,8 @@
 from weave.trace.context.call_context import call_attributes
 from weave.trace.context.tests_context import get_raise_on_captured_errors
 from weave.trace.errors import OpCallError
-from weave.trace.op_extensions.log_once import log_once
 from weave.trace.refs import ObjectRef
+from weave.trace.util import log_once
```
Comment (andrewtruong) [inline weave/trace/op.py:33]: see comment below

File: weave/trace/op_extensions/accumulator.py [hunk 3]
```diff
@@ -10,7 +10,7 @@
 
 from weave.trace.context.tests_context import get_raise_on_captured_errors
 from weave.trace.op import FinishCallbackType, Op
-from weave.trace.op_extensions.log_once import log_once
+from weave.trace.util import log_once
```
Comment (andrewtruong) [inline weave/trace/op_extensions/accumulator.py:13]: see comment below

File: weave/trace/weave_client.py [hunk 4]
```diff
@@ -41,9 +42,9 @@
 from weave.trace.serializer import get_serializer_for_obj
 from weave.trace.settings import client_parallelism
 from weave.trace.table import Table
-from weave.trace.util import deprecated
+from weave.trace.util import deprecated, log_once
```
Comment (andrewtruong) [inline weave/trace/weave_client.py:45]: see comment below

File: weave/trace/weave_client.py [hunk 5]
```diff
@@ -199,6 +202,16 @@ class Call:
...
+    @property
+    def display_name(self) -> str | Callable[[Call], str] | None:
+        return self._display_name
+
+    @display_name.setter
+    def display_name(self, name: str | Callable[[Call], str] | None) -> None:
+        if isinstance(name, str):
+            name = elide_display_name(name)
+        self._display_name = name
+
```
Comment (andrewtruong) [inline weave/trace/weave_client.py:214]: this is kinda ugly, but it will ensure the `display_name` is always correct

**Review by tssweeney — APPROVED** [review]

**Pull request merged**